argocd-client/provider.tf

```diff
@@ -0,0 +1,48 @@
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.cluster[0].endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster[0].certificate_authority[0].data)
+
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    command     = "aws"
+    # This requires the awscli to be installed locally where Terraform is executed
+    args = ["eks", "get-token", "--cluster-name", var.cluster_name]
+  }
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.cluster[0].endpoint
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster[0].certificate_authority[0].data)
+
+    exec {
+      api_version = "client.authentication.k8s.io/v1beta1"
+      command     = "aws"
+      # This requires the awscli to be installed locally where Terraform is executed
+      args = ["eks", "get-token", "--cluster-name", var.cluster_name]
+    }
+  }
+}
+provider "kubectl" {
```

Comment:
do we need kubectl provider here?

Reply:
It's my mistake, we don't use kubectl in this module.
Remove it